src/test/scala/com/databricks/labs/mosaic/expressions/geometry/ST_UnionTest.scala

```diff
@@ -0,0 +1,62 @@
+package com.databricks.labs.mosaic.expressions.geometry
+
+import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI.{ESRI, JTS}
+import com.databricks.labs.mosaic.core.index.{BNGIndexSystem, H3IndexSystem}
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.expressions.CodegenObjectFactoryMode
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class ST_UnionTest extends QueryTest with SharedSparkSession with ST_UnionBehaviors {
+
+    private val noCodegen =
+        withSQLConf(
+          SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "false",
+          SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.NO_CODEGEN.toString
+        ) _
+
+    private val codegenOnly =
+        withSQLConf(
+          SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
+          SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true",
+          SQLConf.CODEGEN_FACTORY_MODE.key -> CodegenObjectFactoryMode.CODEGEN_ONLY.toString
+        ) _
+
+    test("Testing stUnion (H3, JTS) NO_CODEGEN") { noCodegen { unionBehavior(H3IndexSystem, JTS) } }
+    test("Testing stUnion (H3, ESRI) NO_CODEGEN") { noCodegen { unionBehavior(H3IndexSystem, ESRI) } }
+    test("Testing stUnion (BNG, JTS) NO_CODEGEN") { noCodegen { unionBehavior(BNGIndexSystem, JTS) } }
```

Comment:
No need to test this over multiple grid systems.
Please remove all BNG tests, and test only for H3. Or even better: create a new index system (under tests, not src) called `MockIndexSystem` that implements the trait and throws unimplemented exception on all methods. 
This can then be used for all the ST_ functions tests (better to do this in a separate PR).

Reply:
Great idea! Lots of tests are suffering from this flaw. Will take a stab at this later this week.